### PR TITLE
Feature/create service simple

### PR DESCRIPTION
**Pull request by trainb0t**
This is a UX feature which helps users creating a service for the first time(or after a long time) in Environment Manager. It the culmination of all the setup for `deployment maps`, `services`, `roles`, `upstreams` and `load balancer` settings in one easy to use wizard with some helpful links. 

Here are some screenshots to give you an example of the UI. 

### Step 1

Basic service details for an HTTP service. 

image

### Step 2

Deployment maps, environment and server role. 

image

### Step 3 

Confirmation of details.

image

### Step 4

All configs done (load balancer and upstreams included). 

image

### CODE REVIEW

File: client/app/deploy/DeployController.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        vm.model.DeploymentMaps[index].SelectedRoleFullName = `NAME:${result.selectedServerRoleName},PLATFORM:${result.selectedPlatform},TYPE:${result.selectedPlatformSize}`;
+        vm.model.DeploymentMaps[index].SelectedRole = `${result.selectedServerRoleName}`;
+      });
+    }
+
+    vm.removeDeploymentMap = function (index) {
+      vm.model.DeploymentMaps.splice(index, 1);
+    }
+
+    vm.doAllTheThings = function () {
```
Comment (DavidIHunt) [inline client/app/deploy/DeployController.js]: Lol... Do ALL the things!!

File: client/app/deploy/DeployController.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            createLoadBalancerSettings()
+          ]);
+        })
+        .then(function () { vm.result = completedJobs; $scope.$apply(); console.log(vm.result) });
+    };
+
+    vm.finishedWizard = function () {
+    };
+
+    vm.cancelledWizard = function () {
```
Comment (DavidIHunt) [inline client/app/deploy/DeployController.js]: Can vm.finishedWizard and vm.cancelledWizard be deleted?

Comment (trainb0t) [reply client/app/deploy/DeployController.js]: Yes sure

File: client/app/deploy/ServerRoleController.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* TODO: enable linting and fix resulting errors */
```
Comment (DavidIHunt) [inline client/app/deploy/ServerRoleController.js]: Outstanding todo

Comment (trainb0t) [reply client/app/deploy/ServerRoleController.js]: Yeah, I think we need a plan for this. I am really not feeling these es-lint rules.

File: client/app/deploy/services/loadBalancerService.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/* TODO: enable linting and fix resulting errors */
```
Comment (DavidIHunt) [inline client/app/deploy/services/loadBalancerService.js]: Outstanding todo

Comment (trainb0t) [reply client/app/deploy/services/loadBalancerService.js]: Yeah, I think we need a plan for this. I am really not feeling these es-lint rules.

File: client/app/deploy/services/loadBalancerService.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+/* eslint-disable */
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+
+'use strict';
+
+angular.module('EnvironmentManager.deploy').service('clientLoadBalancerService',
+  function ($http) {
+    var self = this;
+
+    self.dnsSuffix = 'service.ttlnonprod.local';
```
Comment (DavidIHunt) [inline client/app/deploy/services/loadBalancerService.js]: This is trainline specific.

File: client/app/deploy/services/loadBalancerService.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        };
+        return $http.post(`/api/v1/config/lb-settings`, loadBalancerSettings);
+      });
+    };
+
+    self.getDnsSuffix = function (environment) {
+      if (environment === 'pr1') {
+        return 'service.ttlprod.local';
+      }
+      return 'service.ttlnonprod.local';
```
Comment (DavidIHunt) [inline client/app/deploy/services/loadBalancerService.js]: More trainline specific stuff.

File: client/app/deploy/ServerRoleController.js (hunk 3 above)
Comment (trainb0t) [reply client/app/deploy/ServerRoleController.js]: > Line 84 exceeds the maximum line length of 150
> EOL at end of file
> No blank link at start of scope

Some of this stuff is silly.

Comment (trainb0t) [reply client/app/deploy/ServerRoleController.js]: Not trailing spaces ... the list just goes on and on.

File: client/app/deploy/services/serverRoleService.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      create: create
+    };
+
+    function reduceDeploymentMaps(model) {
+      return model.DeploymentMaps.reduce(function (acc, curr) {
+        acc[curr.SelectedName] = acc[curr.SelectedName] || [];
+        acc[curr.SelectedName].push(curr);
+        return acc;
+      }, {});
+    }
```
Comment (DavidIHunt) [inline client/app/deploy/services/serverRoleService.js]: Recreation of groupby... `return _.groupBy(sampleMaps, 'SelectedName');`

File: server/package.json.orig [hunk 8]
```diff
@@ -0,0 +1,127 @@
+{
```
Comment (DavidIHunt) [inline server/package.json.orig]: Presumably this file can be deleted before we merge.

File: server/.eslintrc.js [hunk 9]
```diff
@@ -3,37 +3,7 @@ module.exports = {
     "ecmaVersion": 6,
     "sourceType": "script"
   },
-  "extends": "airbnb-base",
   "plugins": [
-    "dependencies",
-  ],
-  "rules": {
```
Comment (DavidIHunt) [inline server/.eslintrc.js]: Whoops... Server side eslint rules have disappeared completely. Was that intentional?

**Comment by trainb0t** [comment]
> No tests?

I was thinking of adding some silenium UI tests. We already have a suite of web driver.io tests that I started in the root of the project.

File: client/app/deploy/services/loadBalancerService.js (hunk 5 above)
Comment (trainb0t) [reply client/app/deploy/services/loadBalancerService.js]: How do we usually deal with things like this?

**Review by DavidIHunt — DISMISSED** [review]

File: client/app/deploy/services/serverRoleService.js (hunk 7 above)
Comment (trainb0t) [reply client/app/deploy/services/serverRoleService.js]: Yup worked!

File: client/app/configuration/services/permissionsValidationService.spec.js [hunk 10]
```diff
@@ -91,11 +91,6 @@ describe('permissions validation', function () {
     expect(errors).toContain('Unknown attribute "Rubbish" on permission 1');
   });
 
-  it('should not throw given parsing errors', function () {
```
[the remark this answers is not in the document]
Comment (trainb0t) [reply client/app/configuration/services/permissionsValidationService.spec.js]: I fixed the error message being asserted on and put the test back. 

`An error occurred evaluating permission 1` -> `Unknown attribute "0" on permission 1`

File: server/.eslintrc.js (hunk 9 above)
Comment (trainb0t) [reply server/.eslintrc.js]: Yes, it was intentional. I put them back as I am happy with the relaxed rules on the client.

**Review by DavidIHunt — APPROVED** [review]

**Comment by trainb0t** [comment]
Closing this out, we have moved house!

**Pull request closed without merging**